frontend/src/metabase-types/api/collection.ts

```diff
@@ -127,9 +128,28 @@ export type ListCollectionItemsRequest = {
   pinned_state?: "all" | "is_pinned" | "is_not_pinned";
   sort_column?: "name" | "last_edited_at" | "last_edited_by" | "model";
   sort_direction?: "asc" | "desc";
+  namespace?: "snippets";
 } & PaginationRequest;
 
 export type ListCollectionItemsResponse = {
   data: CollectionItem[];
   models: CollectionItemModel[] | null;
 } & PaginationResponse;
+
+export type CollectionRequest = {
+  id: CollectionId;
+};
+
+export type ListCollectionsRequest = {
+  "personal-only"?: boolean;
+};
+
+export type ListCollectionsResponse = Collection[];
+
+export type CreateCollectionRequest = {
+  name: string;
+  description?: string;
+  color?: string; // deprecated
+  parent_id?: CollectionId | null;
```

Comment:
Wrong type that will also lead to errors if used incorrectly.
It can only be `number | null`

Reply:
do you want me to submit a follow up PR or, do you want to handle in https://github.com/metabase/metabase/pull/41825 ?